hana/enable_cost_optimized.sls

```diff
@@ -0,0 +1,58 @@
+{%- from "hana/map.jinja" import hana with context -%}
+{% set host = grains['host'] %}
+
+{% for node in hana.nodes %}
+{% if node.host == host and node.scenario_type is defined and node.scenario_type.lower() == 'cost-optimized' %}
+
+reduce_memory_resources_{{  node.host+node.sid }}:
+    hana.memory_resources_updated:
+      - name: {{  node.host }}
+      {% if node.cost_optimized_parameters is defined %}
+      - global_allocation_limit: {{ node.cost_optimized_parameters.global_allocation_limit }}
+      - preload_column_tables: {{ node.cost_optimized_parameters.preload_column_tables }}
+      {% endif %}
+      - user_name: SYSTEM
+      {% if node.install.system_user_password is defined %}
+      - user_password: {{ node.install.system_user_password }}
+      {% endif %}
+      - sid: {{  node.sid }}
+      - inst: {{  node.instance }}
+      - password: {{  node.password }}
+      - require:
+        - hana_install_{{ node.host+node.sid }}
+{% endif %}
+
+{% if node.host == host and node.secondary is defined %}
```

Comment:
How this if is working? If the node is set as secondary it will apply the changes??
What happens if the node is set as secondary but not as cost-optimized? Will this changes affect the node?

I guess, together of being secondary, cost-optimized flag must there too apply these changes

Reply:
Hi Xabi, yes you are correct. 

here is summary of the logic I wanted:
- check if cost-optimized flag for this node pillar entry
   1) reduce resources on this hana sid
   2) check if this node is secondary - if so apply srHook stuff
-end if

I made a mistake there closing the previous endif early.
I originally intended to put this if condition within the previous condition where I check 'cost-optimized' flag.
The logic will be applied in that case. i need to close the initial condition correctly.
Let me know what you think of this approach, after I fix the above mistake.

Reply:
Yes, this solution looks correct